### PR TITLE
Decomp for aten.dropout

### PR DESCRIPTION
**Pull request by SherlockNoMad**
When exporting dropout with cpu tensor, we get following graph module 
```
    class GraphModule(torch.nn.Module):
        def forward(self, arg0_1: f32[512, 10]):
            empty_memory_format: f32[512, 10] = torch.ops.aten.empty.memory_format([512, 10], dtype = torch.float32, layout = torch.strided, device = device(type='cpu'), pin_memory = False, memory_format = torch.contiguous_format)
            bernoulli_p: f32[512, 10] = torch.ops.aten.bernoulli.p(empty_memory_format, 0.9);  empty_memory_format = None
            div_scalar: f32[512, 10] = torch.ops.aten.div.Scalar(bernoulli_p, 0.9);  bernoulli_p = None
            mul_tensor: f32[512, 10] = torch.ops.aten.mul.Tensor(arg0_1, div_scalar);  arg0_1 = div_scalar = None
            return (mul_tensor,)
```

In addition, if we export with eval() mode, we will have an empty graph. 

However, when exporting with cuda tensor, we got
```
    class GraphModule(torch.nn.Module):
        def forward(self, arg0_1: f32[512, 10]):
            native_dropout_default = torch.ops.aten.native_dropout.default(arg0_1, 0.1, True);  arg0_1 = None
            getitem: f32[512, 10] = native_dropout_default[0];  native_dropout_default = None
            return (getitem,)
``` 
and exporting under eval() mode will still have a dropout node in graph. 

This PR make exporting with CPU tensor also produce aten.native_dropout. 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_refs/nn/functional/__init__.py [hunk 1]
```diff
@@ -176,35 +175,6 @@ def celu(
     return torch.where(a > 0, a, rhs)
 
 
-@register_decomposition(aten.dropout)
-@_inplace_wrapper
-@out_wrapper()
-def dropout(
```
Comment (peterbell10) [inline torch/_refs/nn/functional/__init__.py:182]: If this is just for the sake of export, then wouldn't it make more sense to remove `aten.dropout` from the core decomposition list but still use it in inductor?

Comment (peterbell10) [reply torch/_refs/nn/functional/__init__.py:182]: Oh it's composite. Thats rather annoying.

Still shouldn't delete the reference implementation though, and it's bad for inductor that we're loosing the shortcuts for `not training` and `p in (0, 1)`.

Comment (SherlockNoMad) [reply torch/_refs/nn/functional/__init__.py:182]: ok, I am keeping the ref, but not registering it as decomp.

File: torch/_decomp/decompositions.py [hunk 2]
```diff
@@ -1042,6 +1042,16 @@ def logit_backward(
...
 
 
+@register_decomposition(aten.dropout)
+@aten.dropout.default.py_impl(DispatchKey.CompositeImplicitAutograd)
+@aten.dropout.default.py_impl(DispatchKey.Autograd)
+def dropout(input: Tensor, p: float, train: Optional[bool]):
+    if train and p != 0:
+        return aten.native_dropout(input, p, train)[0]
+    else:
+        return input
```
Comment (ezyang) [inline torch/_decomp/decompositions.py:1052]: I hope this is what the real op does (return the input directly), because this can cause some gnarly bugs if it isn't (especially because you are registering a regular decomp for this op)

**Review by ezyang — APPROVED** [review]

File: torch/_decomp/decompositions.py (hunk 2 above)
Comment (andrewor14) [reply torch/_decomp/decompositions.py:1052]: @bdhirsh I remember you mentioned that `native_dropout` returns a clone in both train and eval branches. Is this OK?

Comment (SherlockNoMad) [reply torch/_decomp/decompositions.py:1052]: both native_dropout_cpu and native_droput_cuda returns a 
```
    output = input.clone();
```

IIUC, this is different from returning the input direction. This is a difference instance of the tensor with different storage. 

I need to update this to `return input.clone()`
